_scripts/app1-4.R

```diff
@@ -1,31 +1,31 @@
 {
-  rm(list=ls())                         # clean out the environment
-  options(show.error.locations = TRUE)  # give line numbers on errors
+  rm(list=ls());                       # clean out the environment
+  options(show.error.locations = TRUE);  # give line numbers on errors
   
-  fish1Weight = 45
-  fish2Weight = 64
-  fish3Weight = 50
-  fish4Weight = 58
-  fish5Weight = 49
+  fish1Weight <- 45;
+  fish2Weight <- 64;
+  fish3Weight <- 50;
+  fish4Weight <- 58;
+  fish5Weight <- 49;
   
   # solve for the mean fish weight
-  meanWeight = fish1Weight + fish2Weight + Fish3weight + fish4Weight + fish5Weight /5
+  meanWeight <- (fish1Weight + fish2Weight + fish3Weight + fish4Weight + fish5Weight) /5;
   
   # solve the variance for each fish
-  fish1Var = (fish1Weight - meanWeight)^2
-  fish2Var = (fish1Weight - meanWeight)^2
-  fish3Var = (fish1Weight - meanWeight)^2
-  fish4Var = (fish1Weight - meanWeihgt)^2
-  fish5Var = (fish1Weight - meanWeight)^2
+  fish1Var <- (fish1Weight - meanWeight)^2;
+  fish2Var <- (fish2Weight - meanWeight)^2;
+  fish3Var <- (fish3Weight - meanWeight)^2;
+  fish4Var <- (fish4Weight - meanWeight)^2;
+  fish5Var <- (fish5Weight - meanWeight)^2;
   
   # find the variance of the sample
-  weightVariance = (fish1Var + fish2Var + fish3Var + fish4Var + fishvar5 / 5)
+  weightVariance <- (fish1Var + fish2Var + fish3Var + fish4Var + fish5Var / 5); # fix variable
```

Comment:
And the semicolon (;) -- very few people who program solely in R use it but I wish they did.  It is definitely good practice to be explicit about the end of a statement.  And if you ever embed C in your code then semicolons are required.  And embedding C in R is very useful if you want your code to run fast (I'll hopefully remember to talk more about this when we get to for loops)